### PR TITLE
S3 Repository: Deprecate remaining `repositories.s3.*` settings

### PR DESCRIPTION
**Pull request by rjernst**
Most of these settings should always be pulled from the repository
settings. A couple were leftover that should be moved to client
settings. The path style access setting should be removed altogether.
This commit adds deprecations for all of these existing settings, as
well as adding new client specific settings for max retries and
throttling.

relates #24143

### CODE REVIEW

File: plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/InternalAwsS3Service.java [hunk 1]
```diff
@@ -75,12 +75,6 @@ public synchronized AmazonS3 client(Settings repositorySettings) {
...
 
-        Integer maxRetries = getValue(repositorySettings, settings,
-            S3Repository.Repository.MAX_RETRIES_SETTING,
-            S3Repository.Repositories.MAX_RETRIES_SETTING);
-        boolean useThrottleRetries = getValue(repositorySettings, settings,
-            S3Repository.Repository.USE_THROTTLE_RETRIES_SETTING,
-            S3Repository.Repositories.USE_THROTTLE_RETRIES_SETTING);
         // If the user defined a path style access setting, we rely on it,
         // otherwise we use the default value set by the SDK
         Boolean pathStyleAccess = null;
```
Comment (dadoonet) [inline plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/InternalAwsS3Service.java]: You wrote in the PR description:

>  The path style access setting should be removed altogether.

But you did not change it. Does it mean that you are planning to do that in another PR?

Comment (rjernst) [inline plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/InternalAwsS3Service.java]: I deprecated it here, and opened #24143 as a reminder to remove it.

File: plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Repository.java [hunk 2]
```diff
@@ -89,41 +89,41 @@
...
          * repositories.s3.canned_acl: The S3 repository supports all S3 canned ACLs : private, public-read, public-read-write,
          * authenticated-read, log-delivery-write, bucket-owner-read, bucket-owner-full-control. Defaults to private.
          */
-        Setting<String> CANNED_ACL_SETTING = Setting.simpleString("repositories.s3.canned_acl", Property.NodeScope);
+        Setting<String> CANNED_ACL_SETTING = Setting.simpleString("repositories.s3.canned_acl", Property.NodeScope, Property.Deprecated);
         /**
          * repositories.s3.base_path: Specifies the path within bucket to repository data. Defaults to root directory.
          */
-        Setting<String> BASE_PATH_SETTING = Setting.simpleString("repositories.s3.base_path", Property.NodeScope);
+        Setting<String> BASE_PATH_SETTING = Setting.simpleString("repositories.s3.base_path", Property.NodeScope, Property.Deprecated);
```
Comment (dadoonet) [inline plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Repository.java]: Ah! I see. I missed that line! Sorry :)

File: plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/InternalAwsS3Service.java (hunk 1 above)
Comment (dadoonet) [inline plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/InternalAwsS3Service.java]: Thanks! https://github.com/elastic/elasticsearch/pull/24144/files#r111885879

**Review by jdconrad — APPROVED** [review]
LGTM

**Comment by rjernst** [comment]
@elasticmachine retest this please
